### PR TITLE
[feature] Add an asynchronous metric jemalloc.profile.active to show whether sampling is currently active 

### PR DESCRIPTION
**Pull request by Unalian**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add an asynchronous metric jemalloc.profile.active to show whether sampling is currently active. This is an activation mechanism in addition to prof.active; both must be active for the calling thread to sample.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [x]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [x]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [x]  Allow: Performance tests
- [x]  Allow: All NOT Required Checks
- [x]  Allow: batch 1, 2 for multi-batch jobs
- [x]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    finally:
+        cluster.shutdown()
+
+
+# Tests that the system.asynchronous_metric_log table gets populated.
+# asynchronous metrics are updated once every 60s by default. To make the test run faster, the setting
+# asynchronous_metric_update_period_s is being set to 2s so that the metrics are populated faster and
+# are available for querying during the test.
+def test_event_time_microseconds_field(started_cluster):
+    node1.query("SYSTEM JEMALLOC ENABLE PROFILE")
```
Comment (antonio2368) [inline tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py:29]: it's enabled initially unless you set `prof_active:false` in `MALLO_CONF`.

File: tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    res_o = node1.query("SELECT * FROM system.asynchronous_metrics WHERE metric ILIKE '%jemalloc.prof.active%' FORMAT Vertical;")
```
Comment (antonio2368) [inline tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py:30]: please add test of the values after disabling profile

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 666daad0469b38b7c17092ddce35a971a1ca1df9 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64842/666daad0469b38b7c17092ddce35a971a1ca1df9/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py (hunk 1 above)
Comment (Unalian) [reply tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py]: > it's enabled initially unless you set `prof_active:false` in `MALLO_CONF`.

Got it. I have added a default case in test.py.

File: tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py (hunk 2 above)
Comment (Unalian) [reply tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py]: > please add test of the values after disabling profile

This case already exists in test.py.

Comment (antonio2368) [reply tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py]: ah sorry, somehow I didn't see it.

**Review by antonio2368 — APPROVED** [review]

**Comment by antonio2368** [comment]
Please fix clang-tidy failures

**Comment by antonio2368** [comment]
the test with asan failed because we don't use jemalloc with them

you can add at the beginning of test

```py
if node.is_built_with_sanitizer():
    pytest.skip(reason: "Disabled for sanitizers")
```

**Comment by Unalian** [comment]
> the test with asan failed because we don't use jemalloc with them
> 
> you can add at the beginning of test
> 
> ```python
> if node.is_built_with_sanitizer():
>     pytest.skip(reason: "Disabled for sanitizers")
> ```

image
Can I just check this box to make the test work?

File: tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    finally:
+        cluster.shutdown()
+
+
+# Tests that the system.asynchronous_metric_log table gets populated.
+# asynchronous metrics are updated once every 60s by default. To make the test run faster, the setting
+# asynchronous_metric_update_period_s is being set to 2s so that the metrics are populated faster and
+# are available for querying during the test.
+def test_event_time_microseconds_field(started_cluster):
```
Comment (antonio2368) [inline tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py:28]: I just noticed that the test is named wrong

Comment (Unalian) [reply tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py:28]: Oh, my fault. You mean the name of the test and the function name in test.py should be the same. Sorry.

Comment (antonio2368) [reply tests/integration/test_asynchronous_metric_jemalloc_profile_active/test.py]: It doesnt have to be same, I just dont understand why it's named like that
You dont test microsecond field

**Pull request merged**